src/payment/strategies/cybersource/cybersource-payment-strategy.ts

```diff
@@ -0,0 +1,145 @@
+import PaymentStrategy from '../payment-strategy';
+
+import { CardinalClient, CardinalOrderData } from './index';
+
+export default class CyberSourcePaymentStrategy implements PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _cardinalClient: CardinalClient
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+                if (!this._paymentMethod || !this._paymentMethod.config) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                return this._cardinalClient.initialize(this._paymentMethod.config.testMode)
+                    .then(() => this._store.getState());
+            });
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.config) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._paymentMethod.config.is3dsEnabled ? this._placeOrderUsing3DS(order, payment, options, this._paymentMethod.clientToken) :
+            this._placeOrder(order, payment, options);
+    }
+
+    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _placeOrderUsing3DS(order: OrderRequestBody, payment: OrderPaymentRequestBody, options?: PaymentRequestOptions, clientToken?: string): Promise<InternalCheckoutSelectors> {
+        if (!clientToken) {
+            return Promise.reject(new MissingDataError(MissingDataErrorType.MissingPaymentMethod));
+        }
+
+        if (!payment.paymentData) {
+            return Promise.reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+        }
+
+        const paymentData = payment.paymentData as CreditCardInstrument;
+
+        return this._cardinalClient.configure(clientToken)
+            .then(() => this._cardinalClient.runBindProcess(paymentData.ccNumber)
```

Comment:
We seem to be missing a bracket here. Otherwise, we need to indent the following line one extra block.

```suggestion
            .then(() => this._cardinalClient.runBindProcess(paymentData.ccNumber))
```